prow/cmd/tot/main.go

```diff
@@ -192,9 +221,55 @@ func (f fallbackHandler) get(jobName string) int {
 	return n
 }
 
+func (f fallbackHandler) getURL(jobName string) string {
+	if f.configAgent == nil {
+		return fmt.Sprintf(f.template, jobName)
+	}
+
+	var spec *pjutil.JobSpec
+	cfg := f.configAgent.Config()
+
+	for _, pre := range cfg.AllPresubmits(nil) {
+		if jobName == pre.Name {
+			spec = new(pjutil.JobSpec)
+			spec.Type = kube.PresubmitJob
+			spec.Job = jobName
+			break
+		}
+	}
+	if spec == nil {
+		for _, post := range cfg.AllPostsubmits(nil) {
+			if jobName == post.Name {
+				spec = new(pjutil.JobSpec)
+				spec.Type = kube.PostsubmitJob
+				spec.Job = jobName
+				break
+			}
+		}
+	}
+	if spec == nil {
+		for _, per := range cfg.AllPeriodics() {
+			if jobName == per.Name {
+				spec = new(pjutil.JobSpec)
+				spec.Type = kube.PeriodicJob
+				spec.Job = jobName
+				break
+			}
+		}
+	}
+	// If spec is still nil, we know nothing about the requested job.
+	if spec == nil {
+		logrus.Errorf("requested job is unknown to prow: %s", jobName)
+		return ""
+	}
+	return fmt.Sprintf("%s/%s", strings.TrimSuffix(f.bucket, "/"), gcs.LatestBuildForSpec(spec))
```

Comment:
I don't feel strong about trimming trailing slashes here - I can invalidate `--fallback-bucket` invocations that include it is another option.